hpke.go

```diff
@@ -147,39 +150,40 @@ type contextParameters struct {
 }
 
 func (cp contextParameters) aeadKey() []byte {
-	return cp.suite.KDF.LabeledExpand(cp.secret, "key", cp.keyScheduleContext, cp.suite.AEAD.KeySize())
+	return cp.suite.KDF.LabeledExpand(cp.secret, cp.suite.hpkeSuiteID(), "key", cp.keyScheduleContext, cp.suite.AEAD.KeySize())
 }
 
 func (cp contextParameters) exporterSecret() []byte {
-	return cp.suite.KDF.LabeledExpand(cp.secret, "exp", cp.keyScheduleContext, cp.suite.KDF.OutputSize())
+	return cp.suite.KDF.LabeledExpand(cp.secret, cp.suite.hpkeSuiteID(), "exp", cp.keyScheduleContext, cp.suite.KDF.OutputSize())
 }
 
 func (cp contextParameters) aeadNonce() []byte {
-	return cp.suite.KDF.LabeledExpand(cp.secret, "nonce", cp.keyScheduleContext, cp.suite.AEAD.NonceSize())
+	return cp.suite.KDF.LabeledExpand(cp.secret, cp.suite.hpkeSuiteID(), "nonce", cp.keyScheduleContext, cp.suite.AEAD.NonceSize())
 }
 
 type setupParameters struct {
 	zz  []byte
 	enc []byte
 }
 
-func keySchedule(suite CipherSuite, mode HPKEMode, zz, info, psk, pskID, pkSm []byte) (contextParameters, error) {
-	err := verifyMode(suite, mode, psk, pskID, pkSm)
+func keySchedule(suite CipherSuite, mode HPKEMode, zz, info, psk, pskID []byte) (contextParameters, error) {
+	err := verifyPSKInputs(suite, mode, psk, pskID)
 	if err != nil {
 		return contextParameters{}, err
 	}
 
-	pskIDHash := suite.KDF.LabeledExtract(nil, "pskID_hash", pskID)
-	infoHash := suite.KDF.LabeledExtract(nil, "info_hash", info)
+	suiteID := suite.hpkeSuiteID()
+	pskIDHash := suite.KDF.LabeledExtract(nil, suiteID, "pskID_hash", pskID)
+	infoHash := suite.KDF.LabeledExtract(nil, suiteID, "info_hash", info)
 
 	contextStruct := hpkeContext{suite.KEM.ID(), suite.KDF.ID(), suite.AEAD.ID(), mode, pskIDHash, infoHash}
```

Comment:
The first three fields of this struct have been killed

Reply:
Great catch! Fixed.

Reply:
Passing!  I had made the same mistake, so I made another pass through https://github.com/cfrg/draft-irtf-cfrg-hpke/pull/127, but I didn't find any discrepancies.